packages/twenty-front/src/modules/workflow/components/WorkflowFormTextFieldInput.tsx

```diff
@@ -0,0 +1,59 @@
+import { TextVariableEditor } from '@/object-record/record-field/form-types/components/TextVariableEditor';
+import { useTextVariableEditor } from '@/object-record/record-field/form-types/hooks/useTextVariableEditor';
+import { WorkflowFormFieldInputBase } from '@/workflow/components/WorkflowFormFieldInputBase';
+import { parseEditorContent } from '@/workflow/search-variables/utils/parseEditorContent';
+import { isDefined } from 'twenty-ui';
+
+type WorkflowFormTextFieldInputProps = {
+  label?: string;
+  defaultValue: string | undefined;
+  placeholder: string;
+  onPersist: (value: null | string) => void;
+  multiline?: boolean;
+  readonly?: boolean;
+};
+
+export const WorkflowFormTextFieldInput = ({
+  label,
+  defaultValue,
+  placeholder,
+  onPersist,
+  multiline,
+  readonly,
+}: WorkflowFormTextFieldInputProps) => {
+  const editor = useTextVariableEditor({
+    placeholder,
+    multiline,
+    readonly,
+    defaultValue,
+    onUpdate: (editor) => {
+      const jsonContent = editor.getJSON();
+      const parsedContent = parseEditorContent(jsonContent);
+
+      onPersist(parsedContent);
+    },
+  });
+
+  if (!isDefined(editor)) {
+    return null;
+  }
+
+  return (
+    <WorkflowFormFieldInputBase
+      label={label}
+      variableMode="full-editor"
+      readonly={readonly}
+      Input={
+        <TextVariableEditor
+          editor={editor}
+          multiline={multiline}
+          readonly={readonly}
+        />
+      }
+      multiline={multiline}
+      onVariableTagInsert={(variable) => {
```

Comment:
Could we avoid writing handlers inline ? We make the effort to write them in a section in the component above the JSX part pretty much everywhere with a coherent naming : handleXYZ